tests/unittests/sources/test_smartos.py

```diff
@@ -412,188 +407,186 @@ def close_transport(self):
+    return legacy_user_dir
 
-    def setUp(self):
-        super(TestSmartOSDataSource, self).setUp()
-
-        self.add_patch(DSMOS + ".get_smartos_environ", "get_smartos_environ")
-        self.add_patch(DSMOS + ".jmc_client_factory", "jmc_cfact")
-        self.legacy_user_d = self.tmp_path("legacy_user_tmp")
-        os.mkdir(self.legacy_user_d)
-        self.add_patch(
-            DSMOS + ".LEGACY_USER_D",
-            "m_legacy_user_d",
-            autospec=False,
-            new=self.legacy_user_d,
-        )
-        self.add_patch(
-            DSMOS + ".identify_file",
-            "m_identify_file",
-            return_value="text/plain",
-        )
 
-    def _get_ds(
-        self,
-        mockdata=None,
-        mode=DataSourceSmartOS.SMARTOS_ENV_KVM,
-        sys_cfg=None,
-        ds_cfg=None,
-    ):
-        self.jmc_cfact.return_value = PsuedoJoyentClient(mockdata)
-        self.get_smartos_environ.return_value = mode
+@pytest.fixture
+def m_jmc_client_factory(mocker):
+    return mocker.patch(
+        DSMOS + ".jmc_client_factory", return_value=PsuedoJoyentClient(None)
+    )
 
-        tmpd = self.tmp_dir()
-        dirs = {
-            "cloud_dir": self.tmp_path("cloud_dir", tmpd),
-            "run_dir": self.tmp_path("run_dir"),
-        }
-        for d in dirs.values():
-            os.mkdir(d)
-        paths = c_helpers.Paths(dirs)
 
-        if sys_cfg is None:
-            sys_cfg = {}
+@pytest.fixture
+def mocks(legacy_user_d, mocker, m_jmc_client_factory):
+    mocker.patch(
+        DSMOS + ".get_smartos_environ",
+        return_value=DataSourceSmartOS.SMARTOS_ENV_KVM,
+    )
+    mocker.patch(
+        DSMOS + ".LEGACY_USER_D",
+        autospec=False,
+        new=legacy_user_d,
+    )
+    mocker.patch(
+        DSMOS + ".identify_file",
+        return_value="text/plain",
+    )
+    mocker.patch("cloudinit.net.activators.subp.subp", return_value=("", ""))
 
-        if ds_cfg is not None:
-            sys_cfg["datasource"] = sys_cfg.get("datasource", {})
-            sys_cfg["datasource"]["SmartOS"] = ds_cfg
 
-        return DataSourceSmartOS.DataSourceSmartOS(
-            sys_cfg, distro=None, paths=paths
-        )
+def _get_ds(paths, ds_cfg=None):
+    sys_cfg = {}
+    if ds_cfg is not None:
+        sys_cfg["datasource"] = {}
+        sys_cfg["datasource"]["SmartOS"] = ds_cfg
+
+    return DataSourceSmartOS.DataSourceSmartOS(
+        sys_cfg, distro=None, paths=paths
+    )
+
+
+@pytest.fixture
+def ds(paths):
+    return lambda: _get_ds(paths)
+
+
+@pytest.mark.usefixtures("fake_filesystem", "mocks")
+class TestSmartOSDataSource:
+    jmc_cfact = None
+    get_smartos_environ = None
 
-    def test_no_base64(self):
+    def test_no_base64(self, paths):
         ds_cfg = {"no_base64_decode": ["test_var1"], "all_base": True}
-        dsrc = self._get_ds(ds_cfg=ds_cfg)
+        dsrc = _get_ds(paths, ds_cfg=ds_cfg)
         ret = dsrc.get_data()
-        self.assertTrue(ret)
+        assert ret
```

Comment:
General question here, do we prefer strict assert on True, to avoid truish behavior passing tests?
```suggestion
        assert ret is True
```
Same question applied throughout. I'd lean toward strict on `assert X is True` . But maybe that's just being pedantic and forcing folks to type more and read more characters.

Reply:
^ not blocking review comment. Just conversation to get your opinion.